docs: first draft of migration doc (#942)
tinacms/tinacms.org: content/docs/tina-cloud/beta-migration.md

---
title: Upgrading from Alpha to Beta
---

## Moving Tina Cloud to Beta

Thanks for participating in the Tina Cloud Alpha! Because of you we were able to gather a ton of useful feedback, identify pain points, 
and get a good idea about how real people will use Tina Cloud in the future.

We've made a few changes and are ready to make a Tina Cloud beta publicly available. 

## Beta Changes

> Aside from some UI changes that you've might have noticed us developing in the Tina Cloud Dashboard, we've also been making a bunch of changes 

Reviewer: ```suggestion
Aside from some user interface improvements in the Tina Cloud Dashboard, we've also been making a bunch of changes 
```

under the hood. One of the biggest of these changes is how we handle user accounts. 

> Previously when you created a Tina Cloud account you were creating an **organization** that you accessed at `https://OrgName.tina.io`. We thought that giving each organization it's own unique subdomain and pool of users would be neat, but after spending some time in that paradigm it turned out to be a lot of added complexity that didn't add much value. 

Reviewer: ```suggestion
Previously when you created a Tina Cloud account you were creating an **organization** that you accessed at `https://OrgName.tina.io`. We thought that giving each organization its own unique subdomain and pool of users would be neat, but after spending some time in that paradigm it turned out to be a lot of added complexity that didn't add much value. 
```

With the release of the beta we're moving user accounts to the forefront. Whenever you create a Tina Cloud account you'll always access it at https://app.tina.io. This means that even
if you are part of multiple organizations, you'll only have a single Tina account.

> Tools for creating/managing organizations within Tina Cloud accounts are coming. For now we create a default organization, which you'll still be able to invite users to.

## Migration Path

> Since we're changing how user accounts work within Tina Cloud, alpha users will need to create new accounts for the beta. The process is pretty simple, you'll just need to **recreate your Tina Cloud account** an **update your Tina Cloud Site**. The good news is that since we don't control your content, there really isn't too much to do. 

Reviewer: ```suggestion
Since we're changing how user accounts work within Tina Cloud, existing alpha users must **recreate their Tina Cloud account** and **update their Tina Cloud Site**. The good news is that since we don't control your content, there really isn't anything else to do. 
```

### Recreate your Tina Cloud Account

* Head to https://app.tina.io and sign up for a new account.

> * Log in to your newly created account [recreate any existing **Apps**](https://tina.io/docs/tina-cloud/dashboard/#apps). 

Reviewer: ```suggestion
* Log in to your newly created account [recreate any existing **Apps**](/docs/tina-cloud/dashboard/#apps). 
```

* Invite any users to your newly created account.

### Update your Tina Cloud site

* Upgrade `tina-graphql-gateway` to `some-version`.
* Update your site's environment variables
  * Delete `ORG_NAME`

>   * Change `APP_ID` to the ne

Reviewer: ```suggestion
  * Change `APP_ID` to the new value
```

That's it! With any luck you should be able to keep working on your Tina enabled site the same as before. We're not expecting to have to repeat this process when we make it to our first full release.

> If you came across any problems or have any questions feel free to head over to our Discord! We'd love to hear from you.

Reviewer: ```suggestion
If you come across any problems or have any questions feel free to [head over to our Discord](https://discord.gg/QC724mKx)! We'd love to hear from you.
```

> Tools for creating/managing organizations within Tina Cloud accounts are coming. For now we create a default organization, which you'll still be able to invite users to.

## Migration Path

Since we're changing how user accounts work within Tina Cloud, existing alpha users must **recreate their Tina Cloud account** and **update their Tina Cloud Site**. The good news is that since we don't control your content, there really isn't anything else to do.

### Recreate your Tina Cloud Account

- Head to https://app.tina.io and sign up for a new account.
- Log in to your newly created account [recreate any existing **Apps**](/docs/tina-cloud/dashboard/#apps).
- Invite any users to your newly created account.

### Update your Tina Cloud site

1. Remove the `tina-graphql-gateway` and `tina-graphql-gateway-cli` dependencies.

```
yarn remove tina-graphql-gateway-cli tina-graphql-gateway
```

2. Add the new Tina dependencies.

```
yarn add tinacms @tinacms/auth
yarn add -D @tinacms/cli
```

3. Change any existing import from `tina-graphql-gateway` to `tinacms`.

4. Update `package.json`. Replace `yarn tina-gql` calls with `yarn tinacms`.

5. Update your site's environment variables.

> - Delete `NEXT_PUBLIC_ORGANIZATION_NAME`

Reviewer 2: In addition to this, do we need to tell the user to remove the orgId from the Client creation?
https://github.com/tinacms/tinacms.org/commit/10d67f4d4caf0682c9d3221ea9a3178291136a9e#diff-732071cf7d2a0299edf199f91126d4343f052eee3948dae5c0f4f6408dc6c769L30

Maybe we add a bullet point to update the Client creation like so: https://tina.io/docs/tina-cloud/client/#the-tinacms-api-client

Reviewer 3: @jamespohalloran Those sections are outdated, most of the new documentation is still in this PR https://github.com/tinacms/tinacms.org/pull/946

Reviewer 2: Ok, so if I'm understanding, we'll still need to add this migration step, but we'll instead link to: https://github.com/tinacms/tinacms.org/pull/946/files#diff-f5b803b1d78f9603c9177621d55c03d2b5ed0371b268d70d9504a76539a48245

Reviewer 3: They don't need to change any of that if they're already set up

Reviewer 2: Ok I'm confused :) The Client API used to take an organizationId, but with v2 it doesn't need to. Shouldn't that change be apart of the migration doc if the user needs to remove that parameter? Or do I have something wrong?

Reviewer 3: Oh I see, yeah they'll want to remove it, but what I was saying was that we don't need them to do all of that stuff since they already have the `Client` set up somewhere, we can just tell them they can remove it where they've referenced it.
